### ISSUE TITLE
phones2phones (cmu -> unicode-ipa) loses syllable separator boundaries

### ISSUE THREAD

**Issue by andrewchidden** [ssb22/lexconvert#1]
Not 100% sure why it’s happening, but [this ugly hack](https://gist.github.com/andrewchidden/9d731cc7a8311990c36e98477a72b28d) I made in a few minutes (which probably breaks other conversions) fixes it.

## Test case:
* Input: `M AH N 0 S T R AA 1 0 S IH 0 T IH Z`
* Expected output: `mʌn.stɹˈɑ.sɪ.tɪz`
* Broken output: `mʌnstɹˈɑsɪtɪz`

## Description
Input is from [Prosodic](https://github.com/quadrismegistus/prosodic) which interops with lexconvert for converting text-to-speech generated phones in roughly the following process:
1. eSpeak generates eSpeak phones
2. Prosodic uses lexconvert to convert eSpeak to CMU
3. Prosodic uses Syllabify to find syllable boundaries in CMU
4. Prosodic adds `0` for CMU syllable boundaries (see test case input example) and tries to use lexconvert to convert CMU to IPA, which strips syllable boundaries.

**Comment by ssb22** [comment]
Thanks.  Fixed in version 0.27, but you'll need to use the new format `unicode-ipa-syls` instead of `unicode-ipa`:

```
$ python lexconvert.py --phones2phones cmu unicode-ipa-syls M AH N 0 S T R AA 1 0 S IH 0 T IH Z
mʌn.stɹˈɑ.sɪ.tɪz
```

This is because the Oxford Dictionary of English (3rd edition 2010) writes IPA without syllable breaks, so I don't think the syllable breaks should be included by default, however I see nothing wrong with creating a variant format that keeps them.

As you found in your gist, the relevant lines are the `('.',syllable_separator,False)` and `(syllable_separator,'',False)` in the `unicode-ipa` data.  The first of these says "if we see a dot (when converting FROM unicode-ipa to something else) then treat it as a syllable separator", and the second says "if we have a syllable separator TO write in unicode-ipa, write it as an empty string".  (The `False` at the end means "this is not a two-way rule".  If your syllable separator is not the empty string then you don't need this `False`; the default is True and makes the rule work for conversion in both directions.  But obviously a rule that works in both directions yet has an empty string on one side of it is going to create confusion, hence the `False`.)

Unfortunately there was also another unrelated bug which affected us here: the setting `stress_comes_before_vowel` was incorrectly applied to syllable separators as well as stress marks.  This is definitely wrong as there is no format in which a syllable separator comes before a vowel at the end of the syllable instead of after it, and the result was confusing to say the least.  It was an easy fix though.

**Comment by andrewchidden** [comment]
Thanks for the update and explanation! Relevant PR on Prosodic’s side: https://github.com/quadrismegistus/prosodic/pull/12
